src/stash/mod.rs

```diff
@@ -351,20 +351,28 @@ impl<V> Stash<V> {
 
     /// Take an item from a slot (if non empty).
     pub fn take(&mut self, index: usize) -> Option<V> {
+        let mut out = None;
```

Comment:
Do you really need *these* changes? IMO, it's enough to make *cleared* stashes act like new stashes (not stashes that happen to become empty).